tests/test_sbt.py

```diff
@@ -586,3 +586,20 @@ def test_save_sparseness(n_children):
             # Leaf nodes can't have children
             if isinstance(node, Leaf):
                 assert all(c.node is None for c in tree_loaded.children(pos))
+
+
+def test_sbt_signatures():
+    factory = GraphFactory(31, 1e5, 4)
+    tree = SBT(factory, d=2)
+
+    sig47 = load_one_signature(utils.get_test_data('47.fa.sig'))
+    sig63 = load_one_signature(utils.get_test_data('63.fa.sig'))
+
+    tree.insert(SigLeaf('47', sig47))
```

Comment:
In the rust side it's taking a signature (`sig47`, in this case) and building the `SigLeaf` internally. ([This][0] is the code that does the conversion).

Issue: with this design you can't set `metadata` (I'm setting it to empty). We don't use metadata that much, but might be good to clarify what is the intended use for it (only internal properties for SBT operations? user-provided metadata?)

[0]: https://github.com/dib-lab/sourmash/blob/3223103f4c04b4cf2463f57da1ddf7c947aa6535/src/index/mod.rs#L239

Reply:
I think all signature metadata should be in `SourmashSignature`. Maybe we can allow `insert` to take hints or something? Dunno. But for now is it OK if I change `tree.insert` to take `SourmashSignature` objects?